docs/src/guides/general/default-branch.md

```diff
@@ -0,0 +1,154 @@
+---
+
+{{< description >}}
+
+Soon you will be able to select your default branch during the project creation steps, but until that time this guide will show you how to manually make the switch yourself. You can complete some of these steps through the management console, but since all can be completed with the CLI those commands alone are listed. Be sure to [install the CLI](http://localhost:1313/development/cli.html#installation) if you have not already done so. It's assumed you are changing the default branch of your project on Platform.sh from `master` to `main`. If using another name for the default branch, update the commands accordingly. 
+
+## New projects
+
+The steps below assume:
+
+- You have an external repository on GitHub, GitLab, or BitBucket you are trying to integrate with Platform.sh.
+- That repository's default branch is not `master`.
+
+### 1. Create a new project
+
+First, create an empty project with the Platform.sh CLI command `platform create`:
+
+```bash
+$ platform create --title='Main Project' --region=us-3.platform.sh --plan=development --environments=3 --storage=5 --no-set-remote
+```
+
+This will create a new project with Master as the parent environment by default. Modify the flags to fit your use case, otherwise exclude and the CLI will ask you to set them individually during creation. Copy the `Project ID` provided when the command completes, and substitute the value in the steps below. You can also visit the provided management console URL for the project (i.e. `https://console.platform.sh/<USER>/<Project ID>`) to verify the steps as you go along.
+
+The project you just created is empty, that is, there is no code initialized on it's Master environment. You will need to have something on that environment to begin with in order to create the `main` branch, so you can initialize it with a template for now:
+
+```bash
+$ platform environment:init -p <Project ID> -e master https://github.com/platformsh-templates/hugo.git
+```
+
+Then, create the "Main" environment:
+
+```bash
+$ platform environment:branch main --title=Main -p <Project ID> --force
+```
+
+{{< note >}}
+The CLI assume that you are running this command within a local copy of your repository, so the `--force` flag is included above to bypass that.
+{{< /note >}}
+
+### 2. Deactivate the Master environment
+
+Since Master is currently the parent environment, it is protected, and so the normal `platform environment:delete` command will not work at this stage. To get around this, place an authenticated cURL request on the project's API with the CLI command to deactivate it:
+
+```bash
+$ platform project:curl -X POST -p <Project ID> environments/master/deactivate 
```

Comment:
We don't document or suggest using `platform:curl` anywhere else at this point.  It's a hack. Is that really as much as we're going to do?

Reply:
For now is my understanding, otherwise they'd have to place the API calls themselves and this is at least cleaner.

Reply:
What about

```sh
platform environment:delete master
```

Reply:
@pjcdawkins I still get this message trying to deactivate master with this command

```
The main environment cannot be deleted.
```

Reply:
Ah that's a bug. The `is_main` property is incorrectly cached on the environment. Run `platform project:curl system/restart -X POST` and you'll see the cache being cleared... and then this CLI command would work as intended.

Obviously that wouldn't be so helpful in documentation but.. do we document broken or work-in-progress things? :man_shrugging:

Reply:
@pjcdawkins So this seems a little inconsistent actually for me, even with the `system/restart`. It worked fine on an eu-3 project, but doesn't work at all on us-3. Any idea when the bug fix will be applied?

Otherwise. Maybe it's good enough for now to keep the `platform project:curl` commands here for now and replace when the fix is in?

cc @Crell

Reply:
```
$ platform project:curl system/restart -X POST -p fw36zqmvvzzls
{
  "status": "OK", 
  "code": 200, 
  "_embedded": {
    "activities": []
  }
$ platform  environment:delete master --no-delete-branch -p fw36zqmvvzzls
The main environment cannot be deleted.
```

Reply:
As long as we know that a better alternative is coming, I'm OK with showing `project:curl` for now, provided it's marked as non-standard.  But that should be seen as extra incentive to finish the better alternative, not an excuse not to. :smile: